Fix HOC header so that links always fits the bar (even with long strings)

kozzi opened the pull request:
Regressed by #6262:
### Before

![image](https://cloud.githubusercontent.com/assets/1885340/12436205/055df7c8-bec8-11e5-93e9-db2ab0e077e7.png)
### After

![image](https://cloud.githubusercontent.com/assets/1885340/12436214/1083303c-bec8-11e5-8ded-ad17a4df3a8b.png)

Note: Things still break in >768 && <1024 widths, but that is not introduced by #6262. https://hourofcode.com/us/pt is already broken in production right now for 900px wide browsers :(

I can look into that separately, depending on how strongly we feel about fixing immediately. I did check the common tablets and the header looks look fine on portrait ipad, ipad mini and nexus 7 since the exact 768px width will trigger the hamburger menu. 

The case where it screws up is if you're on a device that is >768 and <1024 wide, or if you're on a desktop and decide resize your browser to that width.

I suspect a more general solution requires some dynamic detection of the nav width using javascript (rather than switching on media queries, which is what it uses right now).

kozzi commented:
fyi @tanyaparker

kozzi commented:
@tanyaparker btw, is us/pt the widest/'lengthiest' language we have? Or are there even lengthier ones?

breville commented:
The fix looks okay, but it seems best to consider fixing all languages at all widths while working in this area.  Of course we could use google analytics to decide if some just aren't worth the effort, but if it's not too difficult, it seems the right thing is to get it right and then we don't need to worry about it in the future.

breville commented:
Also worth testing with longer strings, even if we don't have them now, in anticipation of other translations/strings coming in the future.

tanyaparker commented:
Yeah, I agree with Brendan. I don't think it's urgent enough to have a half-fix. I think it's fine to take another day to have a solution for all widths.

Perhaps we can put a "text limit" make it go to hamburger if the total width of the translation is over y pixels?

kozzi commented:
Sounds good! Will look into a more general solution for all widths.

kozzi commented:
Looked at this today. It looks like this is actually not super straightforward. We can't just use the width of the nav menu to determine when to switch to burger since we actually _change_ the width of nav element when we switch to the burger menu ('display: none' has a width of 0 apparently). So it's forever stuck in the burger menu.

Will likely take a bit longer than a day or two to get this in. Now exploring ideas of looking at total character count in the menu to determine the appropriate "breakpoint" for when to switch to burger, since character count does not change on screen width. Still looking... 

Other ideas welcome, if you have any thoughts!

tanyaparker commented:
What if each link had a set width and then we ellipsis past that width? So sometimes the translation won't be complete, but that's something a translator can get feedback for and choose different wording. Better than not fitting on the nav?

breville commented:
I think that makes sense, and is better than trying to rely on character count which would not really be accurate. 

> On Jan 21, 2016, at 9:02 AM, Tanya Parker notifications@github.com wrote:
> 
> What if each link had a set width and then we ellipsis past that width? So sometimes the translation won't be complete, but that's something a translator can get feedback for and choose different wording. Better than not fitting on the nav?
> 
> —
> Reply to this email directly or view it on GitHub.

kozzi commented:
Good idea! Trying that now. It's also a lot simpler.

kozzi commented:
Updated change:
- Added a max-width for each header link, which will ellipsify(?) when string is too long
- Added a tooltip so that a user can still get full text on hover for any clipped string
- Changed the burger threshold from 768px to 992px (bootstrap's "medium" width), cause most languages didn't actually fit under 768px when I was testing. English happened to be one of the few that worked.

![image](https://cloud.githubusercontent.com/assets/1885340/12467522/d77c351c-bf90-11e5-872c-f597f7dfb640.png)
_The screencap doesn't capture tooltips. Strange... but it's there!_

Also, change was verified on IE8 and mobile.

tanyaparker commented:
Super!! I think this is a great solution. lgtm in Chrome, can you verify you tested in IE8-11, Safari, FF, mobile, tablet?

kozzi commented:
Looked good in in IE 8-11, Safari, FF, mobile and tablet!
